### PR TITLE
Adding v3-1 and small changes to examples

### PR DESCRIPTION
**Pull request by Zenrac**
### Checks and guidelines:

* [x] Have you checked that there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Have you checked that only **single quotes** are used in the code, apart from the doc-strings?
* [x] Have you run a lint program on your code prior to submission?
* [x] Have you checked if `python run_tests.py` returns no errors?

Tests output: 
image

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] This change is a documentation update

### Describe the changes:

- Adding `# noqa: W605` because flake8 doesn't like regex.
- Adding `V3-1` example (almost the same thing than V3).
- In the `V3-1` example, I added a `get_player` fonction which is not that useful used like this, but allows to parse guild as argument and it's more user-friendly for (future) modifications.
- I removed `self.bot.lavalink.unregister_hook(self._track_hook)` in every examples because when you re-load the cog, you cannot register the hook anew (as it's registered in the `__init__`). I found it more logical to keep the hook registered in case that the cog gets reloaded. I figured out that the hook can get an event with a `None` player, so I also added a check to prevent it.

### CODE REVIEW

File: examples/music-v3-1.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+            bot.lavalink.nodes.add(lavalink.Regions.all(), password='youshallnotpass')
+            self.bot.lavalink.register_hook(self._track_hook)
+
+    def __unload(self):
+        # Clear the players from Lavalink's internal cache
+        self.bot.loop.create_task(self.bot.lavalink.players.safe_clear())
+
+    async def _track_hook(self, event):
+        if type(event) in [lavalink.Events.StatsUpdateEvent, lavalink.Events.NodeReadyEvent]:
```
Comment (AndyTempel) [inline examples/music-v3-1.py]: Should add `NodeDisabledEvent` event to prevent unnecessary errors.

File: examples/music-v2.py [hunk 2]
```diff
@@ -33,11 +33,12 @@ def __unload(self):
 
         # Clears the players from Lavalink's internal cache
         self.bot.lavalink.players.clear()
-        self.bot.lavalink.unregister_hook(self._track_hook)
```
Comment (AndyTempel) [inline examples/music-v2.py]: I think it's better to also call `self.bot.lavalink.destroy()` to safely disconnect websockets.

File: examples/music-v3-1.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+        if not hasattr(bot, 'lavalink'):
+            lavalink.Client(bot=bot, loop=bot.loop, log_level=logging.DEBUG)
+
+            bot.lavalink.nodes.add(lavalink.Regions.all(), password='youshallnotpass')
+            self.bot.lavalink.register_hook(self._track_hook)
+
+    def __unload(self):
+        # Clear the players from Lavalink's internal cache
+        self.bot.loop.create_task(self.bot.lavalink.players.safe_clear())
```
Comment (AndyTempel) [inline examples/music-v3-1.py]: I think it's better to also call `self.bot.lavalink.destroy()` to safely disconnect websockets.

File: examples/music-v3.py [hunk 4]
```diff
@@ -33,11 +33,12 @@ def __unload(self):
             player.cleanup()
         # Clear the players from Lavalink's internal cache
         self.bot.lavalink.players.clear()
-        self.bot.lavalink.unregister_hook(self._track_hook)
```
Comment (AndyTempel) [inline examples/music-v3.py]: Also here :^) ... call `self.bot.lavalink.destroy()` to safely disconnect websockets.

**Comment by Devoxin** [comment]
To be honest I'm wondering if the V2 and V3 cog in favor of this new cog. Most people should be encouraged to use V3/.1 (this cog looks to be backwards compatible with V3). Thoughts?

File: examples/music-v2.py [hunk 5]
```diff
@@ -33,11 +33,12 @@ def __unload(self):
 
         # Clears the players from Lavalink's internal cache
         self.bot.lavalink.players.clear()
-        self.bot.lavalink.unregister_hook(self._track_hook)
 
     async def _track_hook(self, event):
         if isinstance(event, lavalink.Events.StatsUpdateEvent):
```
Comment (AndyTempel) [inline examples/music-v2.py]: Maybe instead of testing for different events do this: `if not hasattr(event, "player"): return`?
*Applies to all cogs*

**Comment by AndyTempel** [comment]
Also, I'd agree with Devoxin, ditch previous cogs in favour of the new one, since the wrapper is fully compatible with >=2 versions of lavalink and doesn't have any breaking changes if someone uses lavalink v2.

**Comment by Devoxin** [comment]
*cough*loadtracks endpoint*cough*.

Although that being said, perhaps the cog could piggyback on Lavalink's server version detection to adjust result parsing.

**Comment by Devoxin** [comment]
Sounds like a good idea

**Comment by Zenrac** [comment]
So, I merged examples together : 
- Checking if `get_tracks` return a dict, if yes, it's v3+, else v2
- I tried to use `_server_version` but it's better to check the `get_tracks` as @AndyTempel suggested that different versions could be used at the same time.
- I edited the `__unload` and the `__init__` to make lavalink fully destroying itself and can be created anew if the cog is re-loaded.
- Creating two disgusting fonctions `queue_song` and `queue_playing` with way too many args (looking for a better idea)
- Lint again : 
image
(I removed the no-member erros cus wtf)

**Comment by Devoxin** [comment]
fix conflicts when : *^)*

File: examples/music-v3-1.py [hunk 6]
```diff
@@ -11,32 +11,34 @@
...
 
 
 class Music:
     def __init__(self, bot):
         self.bot = bot
 
-        if not hasattr(bot, 'lavalink'):
-            lavalink.Client(bot=bot, password='youshallnotpass',
-                            loop=bot.loop, log_level=logging.DEBUG)
+        if not hasattr(bot, 'lavalink') or (hasattr(bot, 'lavalink') and not bot.lavalink):
```
Comment (Devoxin) [inline examples/music-v3-1.py]:
```suggestion
        if not hasattr(bot, 'lavalink') or not bot.lavalink:
```

**Comment by Zenrac** [comment]
- New command bass boost added
- Fixing conflicts :^)

**Comment by AndyTempel** [comment]
Please be really good boye and fix `./examples/experimental.py:91:1: W293 blank line contains whitespace`

**Comment by Zenrac** [comment]
@Devoxin :grincat:

**Pull request merged**